admin/assets/javascripts/global-scripts.js

```diff
@@ -228,69 +265,163 @@ $(function () {
       }
     },
 
     /**
-     * Find active groups from URL.
-     * @returns {string}
+     * Parses variable from the string provided.
+     * @param variable
      */
-    findActiveGroups: function () {
-      var url = window.location.href
-      if (url.lastIndexOf('/groupIds=') > 0) {
-        return url.substr(url.lastIndexOf('/groupIds=') + 10, url.length)
+    parseQueryString: function (query, variable) {
+      var vars = query.split('&')
+      for (var i = 0; i < vars.length; i++) {
+        var pair = vars[i].split('=')
+        if (decodeURIComponent(pair[0]) === variable) {
+          return decodeURIComponent(pair[1]).replace(/,$/, '')
+        }
       }
+      return ''
     },
 
     /**
-     * Change checkbox status to 'checked' for passed groupIds.
-     * @param groupIds
-     * @returns {boolean}
+     * Gets the window query string and parses it.
+     * @param variable
      */
-    checkGroupCheckbox: function (groupIds) {
-      if (!groupIds) { return false }
-      $('input[name="group"]').prop('checked', false).attr('data-checked', false)
-      $(groupIds).each(function (key, value) {
-        $('input[id="group-' + value + '"]').prop('checked', true).attr('data-checked', true)
-      })
+    getQueryParam: function (variable) {
+      var query = window.location.search.substring(1)
+      return checkboxUtil.parseQueryString(query, variable)
     },
 
     /**
-     * Update sorting link to include active group ids.
-     * @param action
-     * @param groupId
+     * Gets the url from a query string and parses it.
+     * @param variable
      */
-    updateSortingLink: function (action, groupId) {
-      var sortingLink = $('#sortingLink')
-      var insertGroupId = groupId + ','
-      if (sortingLink.length > 0) {
-        if (action === 'add') {
-          if (sortingLink.attr('href').indexOf('groupIds=') === -1) {
-            sortingLink.attr('href', sortingLink.attr('href') + 'groupIds=')
+    getQueryParamFromString: function (querystring, variable) {
+      var searchIndex = querystring.indexOf('?')
+      if (searchIndex === -1) return '' // nothing to be parsed
+
+      return checkboxUtil.parseQueryString(querystring.substring(searchIndex + 1), variable)
+    },
+
+    /**
+    * Re-doing checked state for checkboxes when sorting
+    * @param param
+    */
+    reselectPreviousValues: function (param, updateCountedCheckboxes) {
+      var paramIds = []
+      var activeParamIds = checkboxUtil.getQueryParam(param + 'Ids')
+      if (activeParamIds) {
+        var paramIdsArr = activeParamIds.split(',')
+        paramIdsArr.map(function (p) {
+          paramIds.push(p)
+          checkboxUtil.updateSortingLink('add', param, p)
+        })
+        checkboxUtil.checkCheckbox(param, paramIdsArr)
+
+        var countCheckedCheckboxes = inputStatus.countCheckedCheckboxes()
+        var countAllCheckboxes = inputStatus.countCheckboxes()
+
+        if (updateCountedCheckboxes) {
```

Comment:
Just a minor change here you could avoid the nested statement by returning if the updatedCountedCheckboxes value is not set.